### PR TITLE
First phase of execution strategy layer for #193

### PR DESCRIPTION
**Pull request by vishal-biyani**
Implements a very thin layer on top of pool manager without any change in functionality. The router now interacts with executor and executor, in turn, interacts with pool manager.

---
This change is [Reviewable](https://reviewable.io/reviews/fission/fission/375)

### CODE REVIEW

**Comment by soamvasani** [comment]
Thanks @vishal-biyani. Sorry if this was unclear in the design discussion, but I'm not sure we need an actual new server for this; why not just add a layer to the poolmgr (now executor) service itself? That executor can internally dispatch to the poolmgr as one backend. 

I don't see a lot of advantages in splitting this out as a server+deployment+service.

**Comment by soamvasani** [comment]
Hey Vishal, looks like the tests broke because they still collect logs from a "poolmgr" pod.

File: executor/cleanup.go [hunk 1]
```diff
@@ -14,7 +14,7 @@ See the License for the specific language governing permissions and
 limitations under the License.
 */
 
-package poolmgr
+package executor
```
Comment (soamvasani) [inline executor/cleanup.go]: Hmm, this _is_ a poolmgr-specific file. How about we make a subdirectory under here, and move the poolmgr into it?

File: executor/gp.go [hunk 2]
```diff
@@ -14,7 +14,7 @@ See the License for the specific language governing permissions and
 limitations under the License.
 */
 
-package poolmgr
+package executor
```
Comment (soamvasani) [inline executor/gp.go]: Same comment as above.. we should keep a clear package boundary around the poolmgr, which we could do by moving it into its own package.

**Comment by soamvasani** [comment]
Hey, I'm going to merge #381 before this one, since that one is blocking our compatibility with Kubernetes 1.8. It may be a bit of a big merge conflict, sorry about that.

**Comment by vishal-biyani** [comment]
No issues @soamvasani, CRD is definitely good to have ASAP. I have merged with latest from the master.

**Comment by vishal-biyani** [comment]
Closing this in favor of #384 and more smaller PRs.

**Pull request closed without merging**